### ISSUE TITLE
Seige towers as deployable mobile platforms

### ISSUE THREAD

**Issue by GRhin** [Sh0wdown/UnofficialCrusaderPatch#806]
This suggestion was originally made by carlospaul on the warlords discord:

> would be cool if you could use siege towers as mobile platforms to increase the height of your archers

This has grown on me as seige towers are pretty useless as things stand, they can be countered by moat, assassin proof walls, buildings in front of walls, or simply diagonal walls. After more thought on this way to increase seige towers usefullness I have refined the idea somewhat, let me put everything in bullet points to make it easier to understand:
- Seige towers can be "manned" like a regular tower
- Seige towers can only be "manned" when "deployed"
- To "deploy" a siege tower, you target a piece of land to attack. 
-  Once "deployed" a siege tower is unable to be moved again.
This approach means that you can use seige towers for map control, but not essentially like SHC2s battle wagons. They are still essentially weak, and you cant fit many units in one, so its not OP, but i think it would be a cool way to bring this unit back into usage in the game.

**Comment by Tjapka** [comment]
I would totaly support this case! I felt so betrayed with Stronghold 2 seeing archers on top of siege towers on the cover of the cd box. 
It might be to difficult to do but maybe use the same mechanic the sapperus are using, so siege towers can "drive" just in one line towards a wall once the wall ist attacked?

**Comment by J-T-de** [comment]
This sounds like a complete rewrite of the siege tower functionality. The way the game currently handles siege towers is as follows:
- while moving, the siege tower just behaves like a regular unit
- when attached to a wall, the siege tower unit turns into a siege tower building, which connects the ground with the wall like stairs or ladders

**Comment by GRhin** [comment]
Not entirely correct, as there are ways to levitate units using seige towers. Let me find a video

**Comment by GRhin** [comment]
https://youtu.be/0Y8l9hOlbAw at about 10 till 12 minutes showing them going up (other tricks in video showing in between as well, not a well organized video but good tricks)

**Comment by J-T-de** [comment]
Yeah, thats true, but this is a really weird bug. There is simply a leftover siege tower building at the position, which is not movable.

**Comment by GRhin** [comment]
Exactly, that's precisely what I am suggesting, except keep the graphics of the seige tower there as well. It's hardly a "complete rewrite of seige tower functionality" as we have just shown that all the functionality is already there.
It may not be a simple implementation, but I'll wait for a dev to confirm if it's too much for too little.

**Comment by J-T-de** [comment]
Ah, now it is clear, I am bad at reading. If it is possible, I am not sure if one is able to prevent to stacking units on top of a siege tower like with normal towers, though.

**Comment by GRhin** [comment]
That isn't easy to do. In single player a player can decide if that's OP or not and do or do not use it. In multiplayer most of the tricks either require an attached wall, or time to organize.
Furthermore, by the time towers would be in use for this, shields will be too, which can protect an Archer drawing fire while melee units either climb up to kill them, or simply destroy the tower itself. Therefore these towers are easily countered, giving their use more to protect against raids and slow down an enemy approach, forcing them to use a larger troop composition to combat rather than just running in with horse archers
